job-service/src/main/java/com/hpe/caf/services/job/jobtype/DefaultDefinitionParser.java

```diff
@@ -175,7 +122,11 @@ public void setConfigurationProperties(
                 }
                 // preserve property name case in the output, even though the configuration lookup
                 // ignores it
-                configuration.put(propertyName, propertyValue);
+                if (propertyName.equals("TARGET_PIPE") && propertyValue.equals("")) {
+                    configuration.put(propertyName, null);
```

Comment:
I think it's ok. The original code had this logic in this same class (although in a different method, `getTargetPipe` which has now been removed):

https://github.com/JobService/job-service/blob/develop/job-service/src/main/java/com/hpe/caf/services/job/jobtype/DefaultDefinitionParser.java#L149

```
public String getTargetPipe(final String id, final AppConfig appConfig)
            throws InvalidJobTypeDefinitionException
{
    final String targetPipe = appConfig.getJobTypeProperty(id, "target_pipe");
    if (targetPipe == null) {
        throw new InvalidJobTypeDefinitionException(id + ": target pipe is not configured");
    } else if (targetPipe.equals("")) {
        return null;
    } else {
        return targetPipe;
    }
}
```

I could move it to somewhere else such as the [JsltTaskBuilder ](https://github.com/JobService/job-service/blob/SCMOD-12505/job-service/src/main/java/com/hpe/caf/services/job/jobtype/JsltTaskBuilder.java) or [JobType](https://github.com/JobService/job-service/blob/SCMOD-12505/job-service/src/main/java/com/hpe/caf/services/job/jobtype/JobType.java) class, but at that point, we're working with a `JsonNode` rather than a string, so it's slightly more awkward (have to check if JsonNode has prop, have it check its type, cast it in order to set it again).